libminifi/include/utils/GeneralUtils.h

```diff
@@ -174,6 +174,21 @@ auto invoke(F&& f, Args&&... args) MINIFICPP_UTIL_DEDUCED(detail::invoke_impl(st
 using std::invoke
 #endif /* < C++17 */
 
+#if __cplusplus < 201703L
+#define CPP17_INLINE
+#else
+#define CPP17_INLINE inline
+#endif /* < C++17 */
+
+namespace detail {
+struct dereference_t {
+  template<typename T>
+  T &operator()(T *ptr) const noexcept { return *ptr; }
+};
+}  // namespace detail
+
+CPP17_INLINE constexpr detail::dereference_t dereference{};
```

Comment:
why do we need the `inline`, don't const variables have internal linkage?

Reply:
I didn't think about it, just followed a common pattern. Apparently inline variables have external linkage, so translation units share a single instance.
https://wiki.tum.de/download/attachments/93291100/Riedel%20report%20-%20Constexpr%20and%20Inline%20Variables%20in%20C%2B%2B17.pdf?version=1&modificationDate=1532345169027&api=v2
https://www.learncpp.com/cpp-tutorial/global-constants-and-inline-variables/#urvanov-syntax-highlighter-60a2d99848d04196510799

Reply:
the link cites performance reasons for using `inline constexpr` instead of `constexpr`, specifically the memory reserved in each translation unit that includes this header, and the per-TU initialization time, coupled with the fact that now we have different linkage depending on the C++ version, I would say it is not worth it, and would categorize this as premature optimization, I would be curious what others think, or possibly some other non-performance reasons

Reply:
removed `inline` in 0c45ec9